### PR TITLE
Stylelint: Refactored modules a-f for no-descending-specifity rule (#180)

### PR DESCRIPTION
**Pull request by ianmcburnie**
Going to tackle this in several PRs. The ultimate goal is to be able to enable the no-descending-specificy rule for both LESS and CSS. Adhering to this rule keeps our code easier to reason with and means less chance of unexpected specificity clashes/overrides. This PR concentrates on modules A through F. A few of the trickier ones, such as button, combobox and dropdown will need to be done in their own separate PRs and I have added an inline comment to specifically disable those modules from the rule for now.

As grid is no longer in the build system I had to add the disable rule directly to the dist css to stop the CSS linter from complaining.

For card it's not worth us fixing this, given that it's deprecated, so we can just leave the disable rule in place in the source.

Dialog was a bit more interesting in that the LESS was fine, but the DS4 CSS was not, due to the way our DS4 specific rules come after the base rules. So I added an inline disable rule right before the ds4 include. Probably not ideal, so I will revisit this in a follow up PR.

I think we could put this in 7.3.2 to keep things moving, but also happy to wait and do it as 7.4.0 if we wanted.

### CODE REVIEW

File: src/less/button/base/button.less [hunk 1]
```diff
@@ -1,6 +1,8 @@
 @import "../../less/variables-common.less";
 @import "../../less/mixins-common.less";
 
+/* stylelint-disable  no-descending-specificity */
```
Comment (seangates) [inline src/less/button/base/button.less]: I assume we're just punting specificity issues related to these files until we can rework things?

Comment (ianmcburnie) [reply src/less/button/base/button.less]: Yep, I did mention that in the description. The more complex ones we need to come back to separately.

Comment (seangates) [reply src/less/button/base/button.less]: 👍

Comment (ianmcburnie) [reply src/less/button/base/button.less]: My next PR will go through the simple stuff from G through Z (a bit like this PR). Then I will enable the stylelint config rule. Then we will start picking off the complex ones.

**Comment by ianmcburnie** [comment]
Ok. We don't actually have anything else lined up to go in 7.4.0. So it might be weird just doing a release that doesn't actually change our API.

**Comment by seangates** [comment]
True, but it's sometimes easier to follow if PRs are listed together.

Not a big deal.

**Comment by ianmcburnie** [comment]
Hoping to have PR for G through Z done very soon also. The big ones can maybe wait.

**Pull request merged**